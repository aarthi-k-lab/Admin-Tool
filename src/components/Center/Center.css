.center {
  display: flex;
  align-self: center;
  justify-content: center;
}

.center--expand {
  composes: center;
  flex-grow: 1;
}

.center--no-expand {
  composes: center;
  flex-grow: 0;
}
