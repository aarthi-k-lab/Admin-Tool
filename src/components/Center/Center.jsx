import React from 'react';
import PropTypes from 'prop-types';

import './Center.css';

const Center = ({ children, className, disableExpand }) => (
  <div className={className} styleName={Center.getStyles(disableExpand)}>
    { children }
  </div>
);

Center.getStyles = (disableExpand) => {
  if (disableExpand) {
    return 'center--no-expand';
  }
  return 'center--expand';
};

Center.defaultProps = {
  className: '',
  disableExpand: false,
};

Center.propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.string,
  disableExpand: PropTypes.bool,
};

export default Center;
