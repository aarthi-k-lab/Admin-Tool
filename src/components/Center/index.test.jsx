import React from 'react';
import renderer from 'react-test-renderer';
import Center from './Center';

const children = [<div key={1} />];
describe('<Center />', () => {
  test('snapshot test when disabled is false', () => {
    const tree = renderer
      .create(<Center>{children}</Center>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('snapshot test when disabled is true', () => {
    const tree = renderer
      .create(<Center disableExpand>{children}</Center>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
