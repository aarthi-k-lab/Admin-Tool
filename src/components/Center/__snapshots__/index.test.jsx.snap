// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Center /> snapshot test when disabled is false 1`] = `
<div
  className="components-Center-Center__center--expand components-Center-Center__center"
>
  <div />
</div>
`;

exports[`<Center /> snapshot test when disabled is true 1`] = `
<div
  className="components-Center-Center__center--no-expand components-Center-Center__center"
>
  <div />
</div>
`;
